.github/workflows/k8s-apply-sub.yaml

```
@@ -155,8 +155,30 @@ jobs:
           # Check if the force rollout flag is true
           if [ "$FORCE_ROLLOUT" = true ]; then
             echo "::info::INFO - Force rollout restart requested $MICROSERVICE_FULLNAME."
+            
+            # Check if a rollout is already in progress
             if ! kubectl rollout status -n "$K8S_NAMESPACE" "deployment/$MICROSERVICE_FULLNAME" --watch=false >/dev/null 2>&1; then
-              echo "::info::INFO - Rollout already active for $MICROSERVICE_FULLNAME. No force restart needed."
+              echo "::info::INFO - Rollout already active for $MICROSERVICE_FULLNAME. Checking Pods status."
+
+              # Get the status of all Pods
+              PODS_STATUS=$(kubectl get pods -n "$K8S_NAMESPACE" -l app="$MICROSERVICE_FULLNAME" -o jsonpath='{.items[*].status.phase}')
+              PODS_ERROR=false
+
+              # Check if any Pod is in error state
+              for STATUS in $PODS_STATUS; do
+                if [ "$STATUS" == "Error" ] || [ "$STATUS" == "CrashLoopBackOff" ]; then
+                  PODS_ERROR=true
+                  break
+                fi
+              done
+
+              # Force rollout restart if any Pod is in error state
+              if [ "$PODS_ERROR" = true ]; then
```

An alternative would be:
```suggestion
              # Check for Pods in Error state
              ERROR_PODS=$(kubectl get pods -n "$K8S_NAMESPACE" -l app="$MICROSERVICE_FULLNAME" --field-selector=status.phase=Error)
    
              # Check for Pods in CrashLoopBackOff state
              CRASHLOOP_PODS=$(kubectl get pods -n "$K8S_NAMESPACE" -l app="$MICROSERVICE_FULLNAME" --field-selector=status.phase=CrashLoopBackOff)

              # Force rollout restart if any Pod is in Error or CrashLoopBackOff state
              if [ -n "$ERROR_PODS" ] || [ -n "$CRASHLOOP_PODS" ]; then
```